openedx/adg/lms/applications/tests/test_helpers.py

```diff
@@ -380,3 +400,142 @@ def test_get_courses_from_course_groups(user_client, courses):
     catalog_courses = get_courses_from_course_groups([course_group], user)
 
     assert catalog_courses == [courses['test_course1']]
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    'written_application, omni_courses, bu_courses, omni_courses_started, bu_courses_started, congrats, msg, inst',
+    [
+        (False, False, False, False, False, '', '', ''),
+        (
+            True, False, False, False, False, WRITTEN_APPLICATION_COMPLETION_CONGRATS,
+            WRITTEN_APPLICATION_COMPLETION_MSG, ''
+        ),
+        (True, False, False, True, False, '', '', WRITTEN_APPLICATION_COMPLETION_INSTRUCTION),
+        (
+            True, True, False, False, False, PREREQUISITE_COURSES_COMPLETION_CONGRATS,
+            PREREQUISITE_COURSES_COMPLETION_MSG, ''
+        ),
+        (True, True, False, False, True, '', '', PREREQUISITE_COURSES_COMPLETION_INSTRUCTION),
+        (True, True, True, False, False, APPLICATION_SUBMISSION_CONGRATS, APPLICATION_SUBMISSION_INSTRUCTION, ''),
+    ],
+    ids=[
+        'no_step_completed', 'written_app_completed', 'omni_courses_started',
+        'omni_courses_completed', 'bu_courses_started', 'all_steps_completed'
+    ]
+)
+def test_get_application_hub_instructions(
+    written_application,
+    omni_courses,
+    bu_courses,
+    omni_courses_started,
+    bu_courses_started,
+    congrats,
+    msg,
+    inst,
+    test_user
+):
+    """
+    Test the congratulation messages and instructions for each requirement completion till application submission.
+    """
+    application_hub = ApplicationHubFactory(user=test_user)
+
+    if written_application:
+        application_hub.submit_written_application_for_current_date()
+    if omni_courses:
+        application_hub.set_is_prerequisite_courses_passed()
+    if bu_courses:
+        application_hub.set_is_bu_prerequisite_courses_passed()
+
+    actual_messages = get_application_hub_instructions(application_hub, omni_courses_started, bu_courses_started)
+
+    expected_messages = {
+        'congrats': congrats,
+        'message': msg,
+        'instruction': inst
+    }
+
+    assert actual_messages == expected_messages
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    'is_enrolled, is_completed, all_modules_attempted, status, grade, message',
+    [
+        (False, False, False, NOT_STARTED, '', ''),
+        (True, False, False, IN_PROGRESS, '', ''),
+        (True, False, True, RETAKE, '0', RETAKE_COURSE_MESSAGE),
+        (True, True, True, COMPLETED, '100', ''),
+    ],
+    ids=['not_enrolled', 'enrolled_in_course', 'failed_in_course', 'completed_course']
+)
+def test_get_course_card_information_after_enrolling_and_failing_course(
+    is_enrolled, is_completed, all_modules_attempted, status, grade, message, test_user, mocker
+):
+    """
+    Test the course card information when the user has enrolled in a course and when the user has failed the course
```

Comment:
We should be updating this docstring to include all scenarios, like when the user has not enrolled in the course.